### PR TITLE
FIX: perform `agree_and_keep` action only if possible.

### PR DESCRIPTION
**Pull request by vinothkannans**
While deleting spammers from flag modal it's trying to perform `agree_and_keep` action where it's not possible (or already performed).

### CODE REVIEW

**Comment by tgxworld** [comment]
We need a test here :)

File: app/services/user_destroyer.rb [hunk 1]
```diff
@@ -124,8 +124,9 @@ def block_external_urls(user)
   end
 
   def agree_with_flags(user)
+    guardian = Guardian.new(Discourse.system_user)
     ReviewableFlaggedPost.where(target_created_by: user).find_each do |reviewable|
-      reviewable.perform(@actor, :agree_and_keep)
+      reviewable.perform(@actor, :agree_and_keep) if reviewable.actions_for(guardian).has?(:agree_and_keep)
```
Comment (romanrizzi) [inline app/services/user_destroyer.rb:129]: Could this still fail if `@actor` doesn't have enough permissions, even if the system user can agree with the flag? Maybe we should check using `@guardian` instead?

Comment (vinothkannans) [reply app/services/user_destroyer.rb]: it's done.

**Review by romanrizzi — APPROVED** [review]
Looks great. Thanks!

**Pull request merged**